tests/test_crm.py

```diff
@@ -0,0 +1,517 @@
+    keys = tbl.getKeys()
+
+    assert len(keys) == 1
+
+    swVid = keys[0]
+    r = redis.Redis(unix_socket_path=dvs.redis_sock, db=swsscommon.ASIC_DB)
+    swRid = r.hget("VIDTORID", swVid)
+
+    assert swRid is not None
+
+    ntf = swsscommon.NotificationProducer(db, "SAI_VS_UNITTEST_CHANNEL")
+    fvp = swsscommon.FieldValuePairs()
+    ntf.send("enable_unittests", "true", fvp)
+    fvp = swsscommon.FieldValuePairs([(attr, val)])
+    key = "SAI_OBJECT_TYPE_SWITCH:" + swRid
+
+    ntf.send("set_ro", key, fvp)
+
+
+def test_CrmFdbEntry(dvs):
+
+    dvs.runcmd("crm config polling interval 1")
+
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '1000')
+
+    time.sleep(5*60)
+
+    # get counters
+    used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_used')
+    avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_available')
+
+    app_db = swsscommon.DBConnector(swsscommon.APPL_DB, dvs.redis_sock, 0)
+    cfg_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+
+    # create a FDB entry
+    tbl = swsscommon.ProducerStateTable(app_db, "FDB_TABLE")
+    fvs = swsscommon.FieldValuePairs([("port","Ethernet8"),("type","dynamic")])
+    tbl.set("Vlan2:52-54-00-25-06-E9", fvs)
+
+    # create vlan
+    tbl = swsscommon.Table(cfg_db, "VLAN", "|")
+    fvs = swsscommon.FieldValuePairs([("vlanid", "2")])
+    tbl.set("Vlan2", fvs)
+
+    # create vlan member
+    tbl = swsscommon.Table(cfg_db, "VLAN_MEMBER", "|")
+    fvs = swsscommon.FieldValuePairs([("tagging_mode", "untagged")])
+    tbl.set("Vlan2|Ethernet8", fvs)
+
+    # update available counter
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '999')
+
+    time.sleep(1)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_available')
+
+    assert new_used_counter - used_counter == 1
+    assert avail_counter - new_avail_counter == 1
+
+    # update available counter
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_FDB_ENTRY', '1000')
+
+    time.sleep(1)
+
+    # get counters
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_fdb_entry_available')
+
+    assert new_avail_counter == avail_counter
+
+
+def test_CrmIpv4Route(dvs):
+
+    dvs.runcmd("ifconfig Ethernet0 10.0.0.0/31 up")
+
+    dvs.runcmd("crm config polling interval 1")
+
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_IPV4_ROUTE_ENTRY', '1000')
+
+    # add static neighbor
+    dvs.runcmd("ip neigh replace 10.0.0.1 lladdr 11:22:33:44:55:66 dev Ethernet0")
+
+    db = swsscommon.DBConnector(0, dvs.redis_sock, 0)
+    ps = swsscommon.ProducerStateTable(db, "ROUTE_TABLE")
+    fvs = swsscommon.FieldValuePairs([("nexthop","10.0.0.1"), ("ifname", "Ethernet0")])
+
+    time.sleep(5*60)
```

Comment:
I believe the crm config polling interval take effort immediately, do we still need to wait for 300 seconds?